### PR TITLE
Implementing ExecuteSyncCommand.

### PR DESCRIPTION
**Pull request by rajkumar42**
Refactoring kill and kill child processes to UnixUtilities.

### CODE REVIEW

**Comment by rajkumar42** [comment]
@gregg-miskelly @caslan @chuckries @wesrupert

**Comment by gregg-miskelly** [comment]
>                 if (PlatformUtilities.IsLinux() || PlatformUtilities.IsOSX())

This should probably change to 'CanExecuteCommand'

---
Refers to: src/MICore/Debugger.cs:622 in ab47616. [](commit_id = ab47616c64d1a437880f78008e033c488a3411ee, deletion_comment = False)

**Comment by gregg-miskelly** [comment]
>             {

And eliminate this

---
Refers to: src/MICore/Debugger.cs:610 in ab47616. [](commit_id = ab47616c64d1a437880f78008e033c488a3411ee, deletion_comment = False)

File: src/MICore/Transports/ClientServerTransport.cs [hunk 1]
```diff
@@ -66,5 +66,15 @@ public void Send(string cmd)
         {
             _clientTransport.Send(cmd);
         }
+
+        public int ExecuteSyncCommand(string commandDescription, string commandText, int timeout, out string output, out string error)
+        {
+            throw new NotImplementedException();
```
Comment (gregg-miskelly) [inline src/MICore/Transports/ClientServerTransport.cs]: >throw new NotImplementedException(); [](start = 12, length = 36)

these should delegate to _clientTransport

File: src/MICore/Transports/PipeTransport.cs [hunk 2]
```diff
@@ -361,5 +300,66 @@ private void OnProcessExit(object sender, EventArgs e)
...
+                proc.StartInfo.RedirectStandardInput = false;
+                proc.StartInfo.RedirectStandardOutput = true;
+                proc.StartInfo.RedirectStandardError = true;
+                proc.StartInfo.UseShellExecute = false;
+                proc.StartInfo.CreateNoWindow = true;
+                proc.Start();
+                proc.WaitForExit(timeout);
+                exitCode = proc.ExitCode;
+            }
+            finally
```
Comment (gregg-miskelly) [inline src/MICore/Transports/PipeTransport.cs]: >finally [](start = 12, length = 7)

It doesn't seem like this finally block should cover all this. Like if say the CreateNoWindow property setter threw for some crazy reason, we don't want to call this.

Is there any of the methods that you want to call this even if it throws?

**Comment by gregg-miskelly** [comment]
:shipit:

File: src/MICore/Transports/PipeTransport.cs (hunk 2 above)
Comment (rajkumar42) [inline src/MICore/Transports/PipeTransport.cs]: Did some tests, streams are empty when the process did not launch. So there is no reason to read it on finally. 

---
In reply to: [79071150](https://github.com/Microsoft/MIEngine/pull/446#discussion_r79071150) [](ancestors = 79071150)

**Comment by rajkumar42** [comment]
Tested C# and gdb scenarios with VSCode on Linux.

**Pull request merged**